invoices/mixins.py

```diff
@@ -1,166 +1,152 @@
-from django.http import HttpResponse, HttpResponseRedirect
-from django.shortcuts import render, redirect, get_object_or_404
+from django.http import HttpResponse
+from django.shortcuts import get_object_or_404
 from django.conf import settings
-from django.contrib.auth.mixins import PermissionRequiredMixin
-
-from reportlab.pdfbase import pdfmetrics 
-from reportlab.pdfbase.ttfonts import TTFont
-from reportlab.lib.enums import TA_JUSTIFY
-from reportlab.lib.pagesizes import letter
-from reportlab.platypus import SimpleDocTemplate, Paragraph, Spacer, Image
-from reportlab.lib.styles import getSampleStyleSheet, ParagraphStyle
-from reportlab.lib.units import inch
-from reportlab.pdfgen import canvas
-from reportlab.lib.utils import ImageReader
 
-from invoices.models import Invoice
 
+from invoices.models import Invoice
+from reportlab.platypus import Image
+from reportlab.pdfgen import canvas
 from io import BytesIO
 from io import StringIO
-
 from PIL import Image
-
-from datetime import datetime, date 
-import os, errno,io
+from datetime import date 
+import os, errno, io
 
 
 
 class PdfMixin(object):
+    """pdf functionality
+    """
+    def invoice_directory(self,_id, filename):
+        """Directory to save pdf
+        """
+        directory = '{media}{invoices}/{id}/'.format(media=settings.MEDIA_ROOT,invoices='invoices',id=_id)
+        if not os.path.exists(directory):
+            try:
+                os.makedirs(directory)
+            except OSError as e:
+                if e.errno != errno.EXIST:
+                    raise
+        return '{media}{invoices}/{id}/{pdf}'.format(mdia=settings.MEDIA_ROOT,invoices='invoices',id=_id, pdf=filename)
+
+
+    def get_invoice_directory(self,_id, filename):
+        """Get directory pdf file
+        """
+        return 'invoices/{id}/{pdf}'.format(id=_id, pdf=filename)
+
+
+    def save_pdf(self,  **kwargs):
+        """saving invoice pdf
+        """
+        buff = BytesIO()
+        invoice = get_object_or_404(Invoice, pk=kwargs['invoice_id'])
+        filename= str(self.invoice_directory(_id=invoice.client.id,filename='invoice_' + str(invoice)+'.pdf') )
+        image= str(settings.MEDIA_ROOT) + str(self.request.user.logo ) 
+        response = HttpResponse(content_type='application/pdf')
+        response['Content-Disposition'] = 'attachement; filename={0}.pdf'.format('invoice_' + str(invoice)+'.pdf')
+        
+        """Creating pdf layout
```

Comment:
Docstrings are being put at the beginning of a method or class not in between or end